Feature/read only public share

rullzer opened the pull request:
This adds a new option to shares to mark them as read only (currently only for link share). By default this is always false.

* [x] Add db column
* [x] Add to IShare
* [x] Set value for link shares
* [x] Update API to handle for link shares
* [x] Expose in public sharing page

 Todo:
- [x] Add to menu on share sidebar
- [ ] Add to other share types? (email? talk?)
- [x] Hide download button etc on public sharing page
- [x] Extend apps to use this

rullzer committed:
Add db column to store hide download state

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Extend IShare to show hide download state

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Propegate hide download state in share provider

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer committed:
Expose hide download to public page

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

danxuliu commented:
This settings should be called _Disable download_ instead of _Read only_, as _Read only_ sounds like something related to permissions, but you could have a public link that is editable... and _read only_.

rullzer commented:
> This setting should be called Disable download instead of Read only, as Read only sounds like something related to permissions, but you could have a public link that is editable... and read only.

:speak_no_evil: yes... I'll take care of that next week

rullzer commented:
@danxuliu fixed :)

danxuliu committed:
Honour "hide download" in the public share page

When the "hide download" property of a share is set the public share
page will not show the download button nor the menu with the download,
direct link and "Add to your Nextcloud" actions; the "downloadURL"
hidden field will not be included either in the generated HTML.

Despite that, note that the "downloadURL" parameter is still set and
passed to the template, as this could be needed anyway to generate
previews (for example, of audio files).

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Add "Hide download" to the menu of link shares

Hiding the download does not depend on other settings and it does not
affect other settings either (for example, it would be possible to hide
the downloads yet make the share editable), so a simple checkbox was
added to the menu. However, note that this option is only available for
files, but not for folders.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu committed:
Add acceptance tests for hiding download in link shares

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu commented:
Good naming with _Hide download_, much better indeed than _Disable download_; subtle ;-)

I have added a checkbox to the link share menu of files (not folders, as it is not clear how this should work with folders) to hide the download button and header menu in the public share page.

rullzer commented:
Clients: @tobiasKaminsky @marinofaggiana @camilasan and awell @rullzer 

This hideDownload is a new option to modify shares with. (link shares only right now).
It is only required for you to be able to set it.

tobiasKaminsky commented:
@rullzer how is this set? Just like the hidden upload feature?
It is called "Hide download" on UI, but how is the parameter called?

rullzer commented:
> @rullzer how is this set? Just like the hidden upload feature?
> It is called "Hide download" on UI, but how is the parameter called?

the paremeter (whe modifying a share is) `hideDownload`

juliushaertl commented on apps/files_sharing/lib/Controller/ShareAPIController.php:
```diff
@@ -773,7 +776,7 @@ public function updateShare(
 		}
 
-		if ($permissions === null && $password === null && $sendPasswordByTalk === null && $publicUpload === null && $expireDate === null && $note === null) {
+		if ($permissions === null && $password === null && $sendPasswordByTalk === null && $publicUpload === null && $expireDate === null && $note === null && $hideDownload === null) {
```
Any reason we force setting it and not just have a default of 'false' ?

rullzer replied:
We do not force setting it. This is just if nothing is set.

We can't default to false because we do not require to send all the parameters. If it would default to false it would always turn of hide download if for exmple you only change the xpiration date ;)

juliushaertl replied:
:see_no_evil: My mind somehow messed up the if clause, nevermind.

juliushaertl commented:
@rullzer @danxuliu 0xii Setting it in the UI doesn't work for me. The loading spinner is visibile forever and the setting doesn't seem to be set after reload/when opening the share link.

danxuliu commented:
>  Setting it in the UI doesn't work for me. The loading spinner is visibile forever and the setting doesn't seem to be set after reload/when opening the share link.

I fixed that yesterday, but as far as I can see when the branch was squashed by @rullzer my two last commits were left out :-P I will add and squash them again.

rullzer commented:
whoops :S

danxuliu commented:
Done.

tobiasKaminsky commented:
Android implemenetd.

marinofaggiana commented:
@tobiasKaminsky when a file/folder is shared with "hidedownload" option on client the "save/open in" is disable/hide ?

marinofaggiana commented:
https://github.com/nextcloud/ios/issues/729

Add Hide download parameter on Share link, done.

tobiasKaminsky commented:
> @tobiasKaminsky when a file/folder is shared with "hidedownload" option on client the "save/open in" is hide ?

This is completely independent from our clients, but only affects web page, when you open up the share via link.

marinofaggiana commented:
https://github.com/nextcloud/ios/issues/729#issuecomment-441238923
